homepage/views.py

```diff
@@ -18,4 +19,11 @@ def reviews(request):
 
 
 def add_review(request):
-    return render(request, 'homepage/add_review.html')
+    if request.method == "POST":
+        form = ReviewForm(request.POST)
+        if form.is_valid():
+            form.save()
+            return redirect('add_review')

```

Comment:
What happens if it's invalid? We stay on the same page and nothing happens until the user sends valid data?
I also saw you've written a test for valid data with client to return "302" code, but haven't written one for invalid, so I'm not sure what's the logic in that case. Even if nothing happens maybe we should test that or at least add some comments about it

Reply:
If it's invalid when the user push the ```post``` button, a popup window will be shown (It's Django built-in feature)
So I wasn't sure how to test with ```client``` the invalid option, as the popup window is coupled.
We do have a test for it from the Backend.
Maybe a test with the client that tries to post invalid data, and check if the QuerySet size stayed the same before and after, WDYT?

Reply:
In my opinion the way from submitting the form till retrieving back the updated list of object is too long to be the one that determines whether a validation check was applied. Other things can go wrong on the way so passing the test does not necessarily indicates the this Django feature was taking place.
I think it's ok to leave it like that and trust Django, but let's see what the mentors think.